cartographer/mapping/global_trajectory_builder_interface.h

```diff
@@ -58,6 +58,15 @@ class GlobalTrajectoryBuilderInterface {
   virtual void AddSensorData(const sensor::OdometryData& odometry_data) = 0;
   virtual void AddSensorData(
       const sensor::FixedFramePoseData& fixed_frame_pose) = 0;
+
+  // A callback which is called after local SLAM processes an accumulated
+  // 'sensor::RangeData'. If the data was inserted into a submap, reports the
+  // assigned 'NodeId', otherwise 'nullptr' if the data was filtered out.
+  using AccumulatedRangeDataCallback =
+      std::function<void(int /* trajectory ID */, common::Time,
+                         transform::Rigid3d /* local pose estimate */,
+                         sensor::RangeData /* in local frame */,
+                         std::unique_ptr<const mapping::NodeId>)>;
```

Comment:
@ojura @wohe I am currently working on the design for cloud-based SLAM and for that I need to forward all the information necessary to formulate the global SLAM problem in the cloud. Part of that is the InsertionResult. So I am wondering whether we can extend this callback to also include the InsertionResult.

Reply:
Sure, why not.

Reply:
e.g.

`std::function<void(int /* trajectory ID */, MatchingResult)>`

Reply:
Let's see what Wolfgang thinks. I realize I am somewhat late to the party :)

Reply:
No, you're just in time :) Fell free to submit a PR over at [larics/cartographer](https://github.com/larics/cartographer) for the `range-data-callback` branch.

Reply:
@cshuet On the other hand, there doesn't exist a single `MatchingResult` (or `InsertionResult`) class. There are two separate ones, one for 2D and one for 3D local trajectory builder. `GlobalTrajectoryBuilder` is templated. Perhaps changing that belongs in another Cartographer PR.